csvqb/csvqb/tests/unit/writers/test_qbwriter.py

```diff
@@ -597,5 +596,205 @@ def test_about_url_generation_with_multiple_measures():
+        ),
+        QbColumn(
+            "Value",
+            QbSingleMeasureObservationValue(
+                ExistingQbMeasure("http://example.org/existing/measure"),
+                ExistingQbUnit("http://example.org/some/existing/unit"),
+            ),
+        ),
+        QbColumn(
+            "New Attribute",
+            NewQbAttribute.from_data("New Attribute", data["New Attribute"]),
+        ),
+        QbColumn(
+            "Units",
+            QbMultiUnits(
+                [
+                NewQbUnit("Percent", description="unit", parent_unit_uri="http://parent-uri"),
+                NewQbUnit("People", source_uri="http://source-uri")
+                ],
+            ),
+        ),
+    ]
+
+    cube = Cube(metadata, data, columns)
+
+    qbwriter = QbWriter(cube)
+    list_of_new_unit_resources = qbwriter._serialise_unit()
+
+    mapped_uri = {
+        "Pending": {
+            "uri": "some-dataset.csv#attribute/new-attribute/pending",
+            "description": None,
+            "parent_attribute_value_uri": None,
+            "source_uri": None
+        },
+
+        "In Review": {
+            "uri": "some-dataset.csv#attribute/new-attribute/in-review",
+            "description": None,
+            "parent_attribute_value_uri": None,
+            "source_uri": None
+        },
+        "Final": {
+            "uri": "some-dataset.csv#attribute/new-attribute/final",
+            "description": None,
+            "parent_attribute_value_uri": None,
+            "source_uri": None
+        },
+        "Percent": {
+            "uri": "some-dataset.csv#unit/percent",
+            "description": "unit",
+            "parent_unit_uri": "http://parent-uri",
+            "source_uri": None
+        },
+        "People": {
+            "uri": "some-dataset.csv#unit/people",
+            "description": None,
+            "parent_unit_uri": None,
+            "source_uri": "http://source-uri"
+        },
+    }
+    for (label, expected_config) in mapped_uri.items():
+        new_attribute_value = first(list_of_new_unit_resources, lambda x: x.label == label)
+        if label in ("Percent", "People"):
+            assert new_attribute_value is not None
+            assert new_attribute_value.uri_str == expected_config["uri"]
+            assert new_attribute_value.comment == expected_config["description"]
+            assert (expected_config["source_uri"] is None and new_attribute_value.source_uri is None) \
+                   or str(new_attribute_value.source_uri.uri) == expected_config["source_uri"]
+            assert (expected_config[
+                        "parent_unit_uri"] is None and new_attribute_value.parent_attribute_value_uri is None) \
+                   or str(new_attribute_value.parent_attribute_value_uri.uri) == expected_config[
+                       "parent_unit_uri"]
+        else:
+            assert new_attribute_value is None
```

Comment:
Why don't we just remove the items from the dictionary @canwaf if we don't want to test them here?

Reply:
Removed items from the dictionary which are not used for testing in test_serialise_unit function.